src/javascript/binary/base/page.js

```diff
@@ -468,12 +470,58 @@ Header.prototype = {
 
         this.menu.register_dynamic_links();
     },
+    start_clock_ws : function(){
+        var that = this;
+        var clock_handle;
+        var query_start_time;
+        var clock = $('#gmt-clock');
+
+        function init(){
+            BinarySocket.send({ "time": 1});
+            query_start_time = (new Date().getTime());  
+        }
+        BinarySocket.init({
+            onmessage : function(msg){
+                var response = JSON.parse(msg.data);
+                if (response && response.msg_type === 'time') {
+                    responseMsg(response);
+                }
+            }
+        });
+        function responseMsg(response){
+            var start_timestamp = response.time;
+            
+            that.time_now = ((start_timestamp * 1000)+ ((new Date().getTime()) - query_start_time));
```

Comment:
that.time = moment.unix(start_timestamp*1000).add(moment().diff(moment(response. passthrough.client_time),'seconds'),'seconds');

#And continue to work with Moment object